exporter/elasticsearchexporter/README.md

```diff
@@ -16,29 +16,82 @@ This exporter supports sending OpenTelemetry logs and traces to [Elasticsearch](
 
 ## Configuration options
 
-- `endpoints`: List of Elasticsearch URLs. If `endpoints` and `cloudid` are missing, the
-  ELASTICSEARCH_URL environment variable will be used.
-- `cloudid` (optional):
-  [ID](https://www.elastic.co/guide/en/cloud/current/ec-cloud-id.html) of the
-  Elastic Cloud Cluster to publish events to. The `cloudid` can be used instead
-  of `endpoints`.
-- `num_workers` (default=runtime.NumCPU()): Number of workers publishing bulk requests concurrently.
-- `index` (DEPRECATED, please use `logs_index` for logs, `traces_index` for traces): The
-  [index](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices.html)
-  or [data stream](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html)
-  name to publish events to. The default value is `logs-generic-default`.
-- `logs_index`: The
-  [index](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices.html)
-  or [data stream](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html)
-  name to publish events to. The default value is `logs-generic-default`
+Exactly one of the following settings is required:
+
+- `endpoint` (no default): The target Elasticsearch URL to which data will be sent
+  (e.g. `https://elasticsearch:9200`)
+- `endpoints` (no default): A list of Elasticsearch URLs to which data will be sent,
+  attempted in round-robin order
+- `cloudid` (no default): The [Elastic Cloud ID](https://www.elastic.co/guide/en/cloud/current/ec-cloud-id.html)
+  of the Elastic Cloud Cluster to which data will be sent (e.g. `foo:YmFyLmNsb3VkLmVzLmlvJGFiYzEyMyRkZWY0NTY=`)
+
+When the above settings are missing, `endpoints` will default to the
+comma-separated `ELASTICSEARCH_URL` environment variable.
+
+Elasticsearch credentials may be configured via [Authentication configuration][configauth] settings.
+As a shortcut, the following settings are also supported:
+
+- `user` (optional): Username used for HTTP Basic Authentication.
+- `password` (optional): Password used for HTTP Basic Authentication.
+- `api_key` (optional): [Elasticsearch API Key] in "encoded" format.
+
+Example:
+
+```yaml
+exporters:
+  elasticsearch:
+    endpoint: https://elastic.example.com:9200
+    auth:
+      authenticator: basicauth
+
+extensions:
+  basicauth:
+    username: elastic
+    password: changeme
+
+······
+
+service:
+  extensions: [basicauth]
+  pipelines:
+    logs:
+      receivers: [otlp]
+      processors: [batch]
+      exporters: [elasticsearch/log]
```

Comment:
You reference `elasticsearch/log` name here (and `elasticsearch/trace` name below), but only define a single `elasticsearch` exporter instance above. Perhaps worth making the configuration more consistent?

Reply:
Oops, good catch. I'll replace `elasticsearch/log` and `elasticsearch/trace` with a single `elasticsearch` exporter.

Reply:
@carsonip I don't think that's a problem. The persistent queue creates separate storage files for each signal.

See for example with the OTLP/HTTP exporter:

```yaml
exporters:
  otlphttp:
    endpoint: "http://localhost:4318"
    sending_queue:
      storage: file_storage
extensions:
  file_storage:
    directory: .
receivers:
  otlp:
    protocols:
      grpc:
service:
  extensions: [file_storage]
  pipelines:
    logs:
      exporters: [otlphttp]
      receivers: [otlp]
    traces:
      exporters: [otlphttp]
      receivers: [otlp]
```

When you run the collector with this config:

```shell
otelcol-contrib --config config.yaml
```

you'll see two separate storage files created:

```shell
$ ls exporter_otlphttp__* 
exporter_otlphttp__logs  exporter_otlphttp__traces
```

Reply:
On a second thought, definitely worth checking if this is "built-in" behavior of the queue or needs some special handling in the exporter.

Reply:
Nice! Thanks for confirming. Sorry for the confusion.